src/Microsoft.OData.Edm/EdmModelVisitor.cs

```diff
@@ -457,20 +740,17 @@ protected virtual void ProcessElement(IEdmElement element)
 
         protected virtual Task ProcessElementAsync(IEdmElement element)
         {
-            // TODO: DirectValueAnnotationsInMainSchema (not including those in referenced schemas)
-            this.VisitAnnotations(this.Model.DirectValueAnnotations(element));
-
-            return Task.CompletedTask;
+            return this.VisitAnnotationsAsync(this.Model.DirectValueAnnotations(element));
         }
 
         protected virtual void ProcessNamedElement(IEdmNamedElement element)
         {
             this.ProcessElement(element);
         }
 
-        protected virtual async Task ProcessNamedElementAsync(IEdmNamedElement element)
+        protected virtual Task ProcessNamedElementAsync(IEdmNamedElement element)

```

Comment:
why remove async?

Reply:
@xuzhg Same here:

This was recommended in some other PR to return task immediately instead of using `async/await` where possible especially if method perform one async. [Eliding `async/await`](https://dotnetdocs.ir/Post/48/eliding-async-await) is a technique that helps improve performance by overheads that come with `async/await`. The comment https://github.com/OData/odata.net/pull/3006#discussion_r1648585068